run_tests.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+import sys
+import tarfile
+import time
+import urllib
+
+from subprocess import check_call
+
+import start
+
+
+DIR_TEST_ENV = "test/venv"
+DIR_JENKINS_ENV = "jenkins-env"
+VERSION_VIRTUALENV = "13.1.2"
+
+os.environ["JENKINS_HOME"] = "jenkins-master"
+
+
+def check_patches():
+    print "Checking patches"
+    check_call(["./test/check_patches.sh"])
+
+
+class Jenkins(object):
+    def __init__(self):
+        if self._require_venv_setup():
+            print "Running setup"
+            check_call(["./setup.sh", DIR_JENKINS_ENV])
+        print "Starting Jenkins"
+        self.proc = start.start_jenkins()
+
+    def _require_venv_setup(self):
+        if os.path.exists(DIR_JENKINS_ENV) and not os.environ.get('CI'):
+            print "Jenkins environment already exists!"
+            print "Run './setup.sh' if you want to recreate it."
+            return False
+        return True
+
+    def wait_for_started(self):
+        import requests
+
+        max_time = time.time() + 60
+        session = requests.Session()
+
+        while time.time() <= max_time:
+            try:
+                if session.get("http://localhost:8080").status_code == 200:
+                    return
+            except requests.exceptions.ConnectionError:
+                pass
+            time.sleep(0.5)
+        raise Exception("Jenkins did not start successfully - check the logs.")
```

Comment:
You know what? I wonder if we also can use the redo package here. It would make the code way smaller.